expression/builtin_json_vec.go

```diff
@@ -1218,3 +1218,51 @@ func (b *builtinJSONSPrettySig) vecEvalString(input *chunk.Chunk, result *chunk.
 	}
 	return nil
 }
+
+func (b *builtinJSONMergePatchSig) vectorized() bool {
+	return true
+}
+
+func (b *builtinJSONMergePatchSig) vecEvalJSON(input *chunk.Chunk, result *chunk.Column) error {
+	nr := input.NumRows()
+	argBuffers := make([]*chunk.Column, len(b.args))
+	var err error
+	for i, arg := range b.args {
+		if argBuffers[i], err = b.bufAllocator.get(types.ETJson, nr); err != nil {
+			return err
+		}
+		defer func(buf *chunk.Column) {
+			b.bufAllocator.put(buf)
+		}(argBuffers[i])
+
+		if err := arg.VecEvalJSON(b.ctx, input, argBuffers[i]); err != nil {
+			return err
+		}
+	}
+
+	result.ReserveJSON(nr)
+	jsonValues := make([][]*json.BinaryJSON, nr)
```

Comment:
We can make a slice `jsonValue := make([]*json.BinaryJson, len(b.args))`, and each row can use the slice.

Reply:
nice idea!